File: api/src/Domain/Calendar/DataTransferObjects/AppointmentUpdate.cs
using System;
using System.Collections.Generic;

namespace DetailingArsenal.Domain.Calendar {
    public class AppointmentUpdate : IDataTransferObject {
        public Guid ServiceId { get; }
        public Guid ClientId { get; }
        public decimal Price { get; }
        public string? Notes { get; }
        public List<AppointmentBlockUpdate> Blocks { get; set; } = new List<AppointmentBlockUpdate>();

        public AppointmentUpdate(Guid serviceId, Guid clientId, decimal price, string? notes) {
            ServiceId = serviceId;
            ClientId = clientId;
            Price = price;
            Notes = notes;
        }
    }
}

File: app/src/api/scheduling/calendar/data-transfer-objects/appointment-update.ts
import { AppointmentBlockCreate } from '@/api/scheduling/calendar/data-transfer-objects/appointment-block-create';

export type AppointmentUpdate = {
    id: string;
    serviceId: string;
    price: number;
    blocks: AppointmentBlockCreate[];
    clientId: string;
    notes?: string;
};
